packages/headless/src/features/insight-search/insight-result-preview-request-builder.ts

```diff
@@ -0,0 +1,28 @@
+import {getVisitorID} from '../../api/analytics/search-analytics';
+import {
+  HtmlRequest,
+  HtmlRequestOptions,
+} from '../../api/search/html/html-request';
+import {StateNeededByHtmlEndpoint} from '../result-preview/result-preview-request-builder';
+
+export async function buildInsightResultPreviewRequest(
+  state: StateNeededByHtmlEndpoint,
+  options: HtmlRequestOptions,
+  url: string
```

Comment:
Could the `url` be built inside this function instead of the controller? It looks like everything needed is already in the `state`

Reply:
Yeah it got more complicated because to build the url i need the insight configuration ID which is not part of `StateNeededByHtmlEndpoint. I can cast it but it feels a bit sketchy. I'll try a few things.

Reply:
Do you see a better way than:

```typescript
/**
 * Creates an insight `Quickview` controller instance.
 * @param engine - The insight engine.
 * @param props - The configurable `Quickview` properties.
 * @returns A `Quickview` controller instance.
 */
export function buildQuickview(
  engine: InsightEngine,
  props: QuickviewProps
): Quickview {
  const fetchResultContentCallback = () => {
    engine.dispatch(logDocumentQuickview(props.options.result));
  };

  const path = '/quickview';
  const coreProps = {
    options: {
      ...props.options,
      onlyContentURL: true,
    },
  };

  return buildCoreQuickview(
    engine,
    coreProps,
    (state, options) =>
      buildInsightResultPreviewRequest(
        state as StateNeededByInsightHtmlEndpoint,
        options
      ),
    path,
    fetchResultContentCallback
  );
}
```
where
```typescript
export type StateNeededByInsightHtmlEndpoint = StateNeededByHtmlEndpoint &
  InsightConfigurationSection;
```
?

Reply:
yeah, in that case I think you'd change the state needed for the function to include the configuration section.

For the controller, to ensure that you have all the needed reducers (and engine.state has the correct type), I recommend copying the pattern we have in files like `headless-facet.ts` where we call `load*Reducers`